sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/MemberDescriptor.cs

```diff
@@ -35,20 +35,19 @@ internal MemberDescriptor(KnownType containingType, string name)
 
         internal string Name { get; }
 
-        public bool IsMatch(string memberName, ITypeSymbol containingType, bool checkOverriddenMethods) =>
-            memberName == Name &&
-            containingType.Is(ContainingType);
+        public bool IsMatch(string memberName, ITypeSymbol containingType, bool caseInsensitiveComparison) =>
+            containingType.Is(ContainingType) && HasSameName(memberName, Name, caseInsensitiveComparison);
 
-        public bool IsMatch<TSymbolType>(string memberName, Lazy<TSymbolType> memberSymbol, bool checkOverriddenMethods)
+        public bool IsMatch<TSymbolType>(string memberName, Lazy<TSymbolType> memberSymbol, bool caseInsensitiveComparison)
```

Comment:
as a nitpick, you could choose a default value (for example, `bool caseInsensitiveComparison = false`) and then in the CSharp `DeliveringDebugFeaturesInProduction.cs` you could just call `isDevelopmentMethod.IsMatch(methodName, methodSymbol);` and pass the 3rd parameter only for the VBnet implementation